server/sonar-server/src/test/java/org/sonar/server/qualityprofile/ws/RenameActionTest.java

```diff
@@ -50,46 +51,61 @@
   @Rule
   public ExpectedException expectedException = ExpectedException.none();
 
-  private DbClient dbClient = db.getDbClient();
   private String xoo1Key = "xoo1";
   private String xoo2Key = "xoo2";
+  private DbClient dbClient;
   private WsTester tester;
-  private DefaultOrganizationProvider defaultOrganizationProvider = TestDefaultOrganizationProvider.fromUuid("ORG1");
-  private QProfileWsSupport wsSupport = new QProfileWsSupport(dbClient, userSessionRule, defaultOrganizationProvider);
+  private OrganizationDto organization;
+  private RenameAction underTest;
 
   @Before
   public void setUp() {
-    createProfiles();
-
+    dbClient = db.getDbClient();
+    TestDefaultOrganizationProvider defaultOrganizationProvider = TestDefaultOrganizationProvider.from(db);
+    QProfileWsSupport wsSupport = new QProfileWsSupport(dbClient, userSessionRule, defaultOrganizationProvider);
+    underTest = new RenameAction(new QProfileFactory(dbClient), wsSupport, dbClient, userSessionRule);
     tester = new WsTester(new QProfilesWs(
       mock(RuleActivationActions.class),
       mock(BulkRuleActivationActions.class),
-      new RenameAction(new QProfileFactory(dbClient), wsSupport)));
+      underTest));
+    organization = db.organizations().insert();
+
+    createProfiles();
   }
 
   @Test
-  public void rename_nominal() throws Exception {
+  public void rename() {
```

Comment:
I don't see a test ensuring that renaming to a name which exists in another organization is actually supported